x-pack/plugins/observability_solution/observability_ai_assistant/public/utils/get_contextual_insight_messages.ts

```diff
@@ -43,5 +36,12 @@ export function getContextualInsightMessages({
         name: 'get_contextual_insight_instructions',
       },
     },
+    {
```

Comment:
WDYT about adding an `injectContext` parameter to the `/chat/complete` endpoint, that defaults to true, and only is `false` when returning the result of a client-side action?

Reply:
I can do that. 

> when returning the result of a client-side action?

WDYM? Where would I do that?

Reply:
it's in https://github.com/elastic/kibana/blob/1b90076047f9f38728488598e80f9783578ed0a2/x-pack/plugins/observability_solution/observability_ai_assistant/public/service/complete.ts#L38

Reply:
See https://github.com/elastic/kibana/pull/180384/commits/cdd866ef64bff28b47409dbbc423f5c94f1e0c68

I added the `injectContext` but I don't get the logic you suggest for setting `injectContext = false`. If I do that in the linked file, this also applies to insight components which will therefore not have context applied.

Reply:
So by default this parameter should be true, and it should inject a `context` function call into the conversation, by simply appending it to the list. It should only be `false` when the API consumer passes actions, and the LLM requests an action to be executed client side, and the client opens a new connection with the action response appended. We should also be mindful here of API consumers and make it easy for them. Maybe there's another way to do this too, e.g. by injecting context any time we don't see it after a user message without a function response.

Reply:
> It should only be `false` when the API consumer passes actions, and the LLM requests an action to be executed client side, and the client opens a new connection with the action response appended.

Okay, I am not aware how we'd do this and I feel I'm entering a bit of a rabbit hole here. If it's a quick fix and you can provide pointers I'm happy to do it in this PR if not I'll revert the changes that introduced the `injectContext` param and will look into it in a follow-up

Reply:
I think the latter option should be manageable: when complete starts, you check if after the last user message without a function request there has been a `context` function request/response, if not, you add it to the list of messages, and emit the `MessageAdd` event. And then you can remove the logic in `next` around `context`.